scripts/pylib/twister/twisterlib.py

```diff
@@ -3509,12 +3510,16 @@ def calc_one_elf_size(instance):
 
             instance.metrics["handler_time"] = instance.handler.duration if instance.handler else 0
 
-    def add_tasks_to_queue(self, pipeline, build_only=False, test_only=False):
+    def add_tasks_to_queue(self, pipeline, build_only=False, test_only=False, retry_build_errors=False):
         for instance in self.instances.values():
             if build_only:
                 instance.run = False
 
-            if instance.status not in ['passed', 'skipped', 'error']:
+            retry_statuses = ['passed', 'skipped']
+            if not retry_build_errors:
+                retry_statuses.append("error")
+
+            if instance.status not in retry_statuses:
```

Comment:
in line 3530, for those errors due to build failures, we will report it immediately when retry-failed, so if we want to retry those build failures, we should disable those code.
BTW, for code in line 3530, actually it is a minimal bug, because when we do retry option, we already use "results.failed = results.error" to save those build failures as failed, then if we do report again, it will be calculated into failed again, this will make the count of results.failed incorrect.
So the easiest way is to remove those code directly.
```
            # If the instance got 'error' status before, proceed to the report stage
            if instance.status == "error":
                pipeline.put({"op": "report", "test": instance})
```